### PR TITLE
fix(name): adds validation for name and prevents auto populating name…

### PR DESCRIPTION
**Pull request by invincibleJai**
adds validation for name and prevents auto-populating name with .git if in url

Tracks :
- https://jira.coreos.com/browse/ODC-1567

screenshot:

Screenshot 2019-08-08 at 3 44 06 PM

### CODE REVIEW

**Comment by debsmita1** [comment]
Name

Might have to modify the error message @invincibleJai

**Comment by invincibleJai** [comment]
@debsmita1 thanks,  I verified start with number is not allowed so will change the message to 

`Name must consist of lower-case letters, numbers and hyphens. It must start with a letter and end with a letter or number.`

Screenshot 2019-08-08 at 6 40 39 PM

**Comment by invincibleJai** [comment]
@debsmita1 @karthikjeeyar PTAL

File: frontend/packages/dev-console/src/components/import/import-validation-utils.ts [hunk 1]
```diff
@@ -213,5 +222,8 @@ export const detectGitRepoName = (url: string): string | undefined => {
     return undefined;
   }
 
-  return url.split('/').pop();
+  return url
+    .split('/')
+    .pop()
+    .replace('.git', '');
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/import/import-validation-utils.ts]: @invincibleJai Why not replace `.` with `-`? That will make it even work with names with `.` in between.

Comment (invincibleJai) [reply frontend/packages/dev-console/src/components/import/import-validation-utils.ts]: @rohitkrai03 Done

**Pull request merged**